src/layout/IHyprLayout.cpp

```diff
@@ -168,6 +168,7 @@ void IHyprLayout::onWindowCreatedFloating(PHLWINDOW pWindow) {
                     *pWindow->m_realPosition = PARENT->m_realPosition->goal() + PARENT->m_realSize->goal() / 2.F - desiredGeometry.size() / 2.F;
                     pWindow->m_workspace     = PARENT->m_workspace;
                     pWindow->m_monitor       = PARENT->m_monitor;
+                    return;
```

Comment:
this return skips important logic. Since I assume it's to prevent the `*pWindow->m_realPosition = pos;` below, instead you can move it to `else` in each case, or simplify the ifs, or use some boolean. The logic further down (187-193 ...) cannot be skipped.